### ISSUE TITLE
[SOLVED] What are the non-trainable parameters of the model?

### ISSUE THREAD

**Issue by LucasVandroux** [experiencor/basic-yolo-keras#167]
I found the answer to that question but I am posting it here in case someone is asking themselves the same question as it took me some time to figure it out.

_We consider in this example the Tiny Yolo model_

When you are building a model, at the end of the summary, it will give you 3 numbers about the parameters:
```
Total params: 15,840,210
Trainable params: 15,834,098
Non-trainable params: 6,112
```
However, as none of the layers are frozen I was wondering where those non-trainable parameters were coming from. After a bit of digging, I managed to get a list of the non-trainable parameters.
```python
# command
print(self.model.non_trainable_weights)

# output
[<tf.Variable 'norm_1/moving_mean:0' shape=(16,) dtype=float32_ref>,
<tf.Variable 'norm_1/moving_variance:0' shape=(16,) dtype=float32_ref>, 
...,
<tf.Variable 'norm_8/moving_variance:0' shape=(1024,) dtype=float32_ref>]
```
And by summing the shape of the `tf.Variable` in the list you find the number of non-trainable parameters displayed in the model's summary.

You will also notice that the beginning of the name of each of those variables corresponds to the name of one of the `BatchNormalization` layers and that each layer has a `moving_mean` and a `moving_variance`. Indeed, [fchollet](https://github.com/fchollet) is stating himself in a [conversation on an issue](https://github.com/keras-team/keras/issues/4762) in the `keras-team` repository that this is the normal behavior for the `BatchNormalization` layers:

> batchnorm also maintains non-trainable weights, which are updated via layer updates (i.e. not through backprop): the mean and variance vectors.

**TLTR**: The number of none trainable weights of the model comes from the `BatchNormalization` layers whose mean and variance vectors are updated via layer updates instead of backpropagation and therefore are considered as none trainable parameters.

**Comment by hiteshnitetc** [comment]
If some one not defining model definition then self is not necessary. Just a info.
